```bash
# A)FastQC v0.12.1

FASTQC=/LUSTRE/apps/bioinformatica/FastQC_v0.12.1/
export PATH=$PATH:$FASTQC

mkdir -p fastqc

fastqc *.fq.gz -t 24 --nogroup -o ./fastqc &> fastqc.log &

# B) MultiQC v1.10.1

multiqc ./fastqc/*zip -o multiqc

```
## 01.Preprocess

Trimming adapter content

```bash
mkdir -p 01.Preprocess
cd 01.Preprocess

wget https://raw.githubusercontent.com/RJEGR/Cancer_sete_T_assembly/main/TruSeq3-PE-2.fa

sbatch trimmo.sh ../*1.fq.gz

# grep 'Input Read Pairs' slurm-*171018*.err | awk '{print $4, $7, $12, $17, $20}'
# c("Input Read Pairs",	"Both Surviving",	"Forward Only",	"Reverse Only",	"Dropped")

mkdir -p fastqc

fastqc *P.qtrim.fq.gz -t 20 --nogroup -o ./fastqc

```
Create the sample data which contains four cols
Factor 1	Factor 2	Forward	Reverse
```bash

for file in $(ls ../*1.fq.gz | grep fq)
do
basename=${file##*/}
base="${basename%*_*.fq.gz}"
bs="${base%*_CKDL240013335*}"
fwr=${base}_1.P.qtrim.fq.gz
rev=${base}_2.P.qtrim.fq.gz
echo "$bs" "$bs" `printf "$PWD/$fwr"` `printf "$PWD/$rev"`
done > samples.txt


```

The trimmo.sh script contains
```bash
#!/bin/bash
#SBATCH --job-name=trimmo
#SBATCH -N 1
#SBATCH --mem=12GB
#SBATCH --ntasks-per-node=16
#SBATCH -t 6-00:00:00

queries=$1

TRIMM=/LUSTRE/apps/bioinformatica/Trimmomatic-0.40/
trimmomatic="java -jar $TRIMM/trimmomatic-0.40-rc1.jar"

for file in $(ls $queries | grep fq)
do
basename=${file##*/}
base="${basename%*_*.fq.gz}"
infile="${file%*_*.fq.gz}"
$trimmomatic PE -threads 12 -phred33 \
    -summary ${base}-summary.txt \
    ${infile}_1.fq.gz ${infile}_2.fq.gz \
    ${base}_1.P.qtrim.fq.gz ${base}_1.UP.qtrim.fq.gz \
    ${base}_2.P.qtrim.fq.gz ${base}_2.UP.qtrim.fq.gz \
    ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10 SLIDINGWINDOW:4:15 MINLEN:36 LEADING:5 TRAILING:5
done 

exit
```

## 02. Assembly (reference-free denovo assembly)
### Using Trinity
```bash
mkdir -p 02.Assembly
cd 02.Assembly

sbatch Trinity.sh samples.txt

```

The Trinity.sh must contains
```bash

#!/bin/bash
#SBATCH --job-name=Trinity
#SBATCH -N 1
#SBATCH --mem=100GB
#SBATCH --ntasks-per-node=20
#SBATCH -t 6-00:00:00

samples_file=$1 # samples.txt 

# EXPORT=/LUSTRE/apps/bioinformatica/trinityrnaseq-v2.15.1/
# export PATH=$PATH:$EXPORT
# EXPORT=/LUSTRE/apps/bioinformatica/samtools/bin/
# export PATH=$PATH:$EXPORT

Trinity --seqType fq --max_memory 100G --samples_file $samples_file --no_normalize_reads --CPU 20 --output trinity_out_dir

exit
```

### Using RNASpandes

rnaSPAdes is designed primarily for eukaryotic transcriptome assembly from short reads

# [Link1](https://www.rna.uni-jena.de/supplements/assembly/)
# [Link2](https://github.com/ablab/spades/blob/next/docs/rna.md)

```bash

EXPORT=/LUSTRE/apps/bioinformatica/SPAdes-3.15.5-Linux/bin/
export PATH=$PATH:$EXPORT

spades.py -1 left.fastq.gz -2 right.fastq.gz -o output_folder

spades.py --sc --pe1-1 $R1 --pe1-2 $R2 --pe1-fr -t $THREADS -m $MEM --cov-cutoff auto -o $DIR/spades/
rnaspades.py --pe1-1 $R1 --pe1-2 $R2 --pe1-fr -t $THREADS -m $MEM --cov-cutoff auto -o $DIR/rnaspades/


```

### De novo Assembly evaluation
Transrate
rnaQUAST 
BUSCO 
GeneMarkS-T 

Additional reference 
```bash

# A) DNA sequences
# (Centruroides vittatus) https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_030686945.1/
TYPE=GCA
VERSION=1_ASM3068694v1
DATA=GCA_030686945.${VERSION}_genomic.fna.gz

URL=https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/030/686/945/GCA_030686945.1_ASM3068694v1/$DATA

#  -c, resume getting a partially-downloaded file.

wget -c $URL
```