```bash
# A)FastQC v0.12.1

FASTQC=/LUSTRE/apps/bioinformatica/FastQC_v0.12.1/
export PATH=$PATH:$FASTQC

mkdir -p fastqc

fastqc *.fq.gz -t 24 --nogroup -o ./fastqc &> fastqc.log &

# B) MultiQC v1.10.1

multiqc ./fastqc/*zip -o multiqc

```

Trimming adapter content

```bash
#!/bin/bash
#SBATCH --job-name=trimmo
#SBATCH -N 1
#SBATCH --mem=12GB
#SBATCH --ntasks-per-node=16
#SBATCH -t 6-00:00:00

# queryFile=$1

# wget https://raw.githubusercontent.com/RJEGR/Cancer_sete_T_assembly/main/TruSeq3-PE-2.fa

TRIMM=/LUSTRE/apps/bioinformatica/Trimmomatic-0.40/
trimmomatic="java -jar $TRIMM/trimmomatic-0.40-rc1.jar"

for file in $(ls ../*1.fq.gz | grep fq)
do
basename=${file##*/}
base="${basename%*_*.fq.gz}"
infile="${file%*_*.fq.gz}"
$trimmomatic PE -threads 12 -phred33 \
    -summary ${base}-summary.txt \
    ${infile}_1.fq.gz ${infile}_2.fq.gz \
    ${base}_1.P.qtrim.fq.gz ${base}_1.UP.qtrim.fq.gz \
    ${base}_2.P.qtrim.fq.gz ${base}_2.UP.qtrim.fq.gz \
    ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10 SLIDINGWINDOW:4:15 MINLEN:36 LEADING:5 TRAILING:5
done 

exit

```


Additional reference 
```bash

# A) DNA sequences
# (Centruroides vittatus) https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_030686945.1/
TYPE=GCA
VERSION=1_ASM3068694v1
DATA=GCA_030686945.${VERSION}_genomic.fna.gz

URL=https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/030/686/945/GCA_030686945.1_ASM3068694v1/$DATA

#  -c, resume getting a partially-downloaded file.

wget -c $URL
```