
## 02. Assembly (reference-free denovo assembly)
### Using Trinity
```bash
mkdir -p 02.Assembly
cd 02.Assembly

module load trinityrnaseq-v2.15.1

sbatch Trinity.sh samples.txt

```

The Trinity.sh must contains
```bash

#!/bin/bash
#SBATCH --job-name=Trinity
#SBATCH -N 1
#SBATCH --mem=100GB
#SBATCH --ntasks-per-node=20
#SBATCH -t 6-00:00:00

module load trinityrnaseq-v2.15.1

samples_file=$1 # samples.txt 

Trinity --seqType fq --max_memory 100G --samples_file $samples_file --no_normalize_reads --CPU 20

exit
```

## De novo Assembly Metrics
```bash
transrate --assembly soft_filtered_transcripts.fasta
transrate --assembly transcripts.fasta
transrate --assembly Trinity.fasta
```

Several tools, 
### Trinity assessments

https://github.com/trinityrnaseq/trinityrnaseq/wiki/Transcriptome-Contig-Nx-and-ExN50-stats

```bash

util=/LUSTRE/apps/bioinformatica/trinityrnaseq-v2.15.1/util/

$util/TrinityStats.pl Trinity.fasta > TrinityStats.txt

misc=/LUSTRE/apps/bioinformatica/trinityrnaseq-v2.15.1/util/misc/

$misc/get_longest_isoform_seq_per_trinity_gene.pl Trinity.fasta > Trinity.longest.fasta

# This calculation requires that you've first performed transcript abundance estimation, and can then be computed like so:


$misc/contig_ExN50_statistic.pl transcripts.TMM.EXPR.matrix Trinity.fasta transcript  | tee ExN50.transcript.stats

exit

```
CD-HIT can be used to cluster highly similar sequences and retain a single representative sequence per group. Stringent clustering can be done like so: 'cd-hit-est -o cdhit -c 0.98 -i Trinity.fasta -p 1 -d 0 -b 3 -T 10'

### HISAT transcriptome-methods COVERAGE

```bash
sbatch hisat2_build.sh Trinity.fasta # from trinity
sbatch hisat2_build.sh transcripts.fasta # from rnaspades

for i in $(ls *1.P.qtrim.fq.gz); do sbatch hisat_align.sh Trinity.fasta $i; done
for i in $(ls *1.P.qtrim.fq.gz); do sbatch hisat_align.sh transcripts.fasta $i; done

```

### BUSCO
Setup the dataset from BUSCOdb.md. Then, 

```bash
sbatch BUSCO.sh Trinity.longest.fasta
```

The code must contains:

```bash
#!/bin/sh
## Directivas
#SBATCH --job-name=BUSCOpy
#SBATCH -N 1
#SBATCH --mem=100GB
#SBATCH --ntasks-per-node=20
#SBATCH -t 6-00:00:00

# module load python-2.7-anaconda

module load gcc-12.2.0


IN=$1 # Trinity.longest.fasta
out=${IN%.fasta}

EXPORT=/home/rgomez/bin/busco-master/scripts/
export PATH=$EXPORT:$PATH

EXPORT=/LUSTRE/apps/bioinformatica/ncbi-blast-2.14.0+/bin/
export PATH=$EXPORT:$PATH

EXPORT=/LUSTRE/apps/bioinformatica/hmmer-3.3.2/bin/
export PATH=$EXPORT:$PATH

BUSCOdb=/LUSTRE/bioinformatica_data/genomica_funcional/rgomez/BUSCOdb

# Replace database for a loop or if in order to automate the selection of odb10 in the script run_BUSCO.py
 
eukaryota_odb10=$BUSCOdb/eukaryota_odb10
arthropoda_odb10=$BUSCOdb/arthropoda_odb10
metazoa_odb10=$BUSCOdb/metazoa_odb10
mammalia_odb10=$BUSCOdb/mammalia_odb10
bacteria_odb10=$BUSCOdb/bacteria_odb10

CPU=$SLURM_NPROCS

run_BUSCO.py -i $IN -l $arthropoda_odb10 -m transcriptome -o ${out}_arthropoda_odb10 -c $CPU

run_BUSCO.py -i $IN -l $metazoa_odb10 -m transcriptome -o ${out}_metazoa_odb10 -c $CPU

run_BUSCO.py -i $IN -l $eukaryota_odb10 -m transcriptome -o ${out}_eukaryota_odb10 -c $CPU

run_BUSCO.py -i $IN -l $bacteria_odb10 -m transcriptome -o ${out}_bacteria_odb10 -c $CPU

run_BUSCO.py -i $IN -l $mammalia_odb10 -m transcriptome -o ${out}_mammalia_odb10 -c $CPU

exit
```

Entonces generamos las figuras

```bash
mkdir summaries
cp ./run_*odb*/short_summary* summaries

module load R-4.3.1

EXPORT=/home/rgomez/bin/busco-master/scripts/
export PATH=$EXPORT:$PATH

generate_plot.py --working_directory ./summaries/

cp summaries/busco_figure.R .
sed -i 's/_odb10//g' summaries/busco_figure.R
Rscript summaries/busco_figure.R
# firefox summaries/busco_figure.png
cp summaries/busco_figure.R .
```

Tambien retenemos contigs en los que su status de identidad resulto `completo`, `duplicado` y `fragmentado`.


### transrate
```bash
#!/bin/sh
## Directivas
#SBATCH --job-name=rate
#SBATCH -N 1
#SBATCH --mem=100GB
#SBATCH --ntasks-per-node=20
#SBATCH -t 6-00:00:00

EXPORT=/LUSTRE/apps/bioinformatica/ruby2/ruby-2.2.0/bin
export PATH=$PATH:$EXPORT

TRANSRATE=/LUSTRE/apps/bioinformatica/ruby2/ruby-2.2.0/bin
export PATH=/LUSTRE/apps/bioinformatica/.local/bin:$PATH
export PATH=/LUSTRE/apps/bioinformatica/ruby2/ruby-2.2.0/bin:$PATH
export LD_LIBRARY_PATH=/LUSTRE/apps/bioinformatica/ruby2/ruby-2.2.0/lib:$LD_LIBRARY_PATH

transrate --assembly Trinity.fasta --left tmp.1.P.qtrim.fq.gz --right tmp.2.P.qtrim.fq.gz --threads $SLURM_NPROCS --output transrate_dir

exit


```

### Using RNASpades

rnaSPAdes is designed primarily for eukaryotic transcriptome assembly from short reads

# [Link1](https://www.rna.uni-jena.de/supplements/assembly/)
# [Link2](https://github.com/ablab/spades/blob/next/docs/rna.md)

Using files from 01.Preprocessing step
```bash
ln -s /LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Centruroides/01.Preprocess/*.P.qtrim.fq.gz .

R1=A11_CKDL240013335-1A_22JMN3LT3_L3_1.P.qtrim.fq.gz
R2=A11_CKDL240013335-1A_22JMN3LT3_L3_2.P.qtrim.fq.gz

sbatch spades.sh $R1 $R2
```
The sh file must contain:

```bash
#!/bin/bash
#SBATCH --job-name=spades
#SBATCH -N 1
#SBATCH --mem=100GB
#SBATCH --ntasks-per-node=20
#SBATCH -t 6-00:00:00

EXPORT=/LUSTRE/apps/bioinformatica/SPAdes-3.15.5-Linux/bin/
export PATH=$PATH:$EXPORT

R1=$1
R2=$2

THREADS=$SLURM_NPROCS
MEM=$SLURM_MEM_PER_NODE
DIR=$SLURM_SUBMIT_DIR

# spades.py -1 left.fastq.gz -2 right.fastq.gz -o output_folder

rnaspades.py --pe1-1 $R1 --pe1-2 $R2 --pe1-fr -t $THREADS -m $MEM --cov-cutoff auto -o $DIR/rnaspades/

# https://home.cc.umanitoba.ca/~psgendb/doc/spades/manual.html
# Specifying multiple libraries
# Paired-end libraries


ls *qtrim.fq.gz > FastqPaired.txt

FastqPaired=FastqPaired.txt

N=( $(cat FastqPaired) )
NSZ=${#N[@]}

for((i=2; i<NSZ; i+=2)); do
    
    R1=${N[$i]}
    
    R2=${N[$(($i+1))]}
    
    # OUTPREFIX=${N[$i]%%_*}

    echo "--pe1-1 $R1 --pe1-2 $R2"
done


rnaspades.py \
--pe1-1 A11_CKDL240013335-1A_22JMN3LT3_L3_1.P.qtrim.fq.gz --pe1-2 A11_CKDL240013335-1A_22JMN3LT3_L3_2.P.qtrim.fq.gz \
--pe1-1 A12_CKDL240013335-1A_22JMN3LT3_L3_1.P.qtrim.fq.gz --pe1-2 A12_CKDL240013335-1A_22JMN3LT3_L3_2.P.qtrim.fq.gz \
--pe1-1 A13_CKDL240013335-1A_22JMN3LT3_L3_1.P.qtrim.fq.gz --pe1-2 A13_CKDL240013335-1A_22JMN3LT3_L3_2.P.qtrim.fq.gz \
--pe1-1 D11_CKDL240013335-1A_22JMN3LT3_L3_1.P.qtrim.fq.gz --pe1-2 D11_CKDL240013335-1A_22JMN3LT3_L3_2.P.qtrim.fq.gz \
--pe1-1 D12_CKDL240013335-1A_22JMN3LT3_L3_1.P.qtrim.fq.gz --pe1-2 D12_CKDL240013335-1A_22JMN3LT3_L3_2.P.qtrim.fq.gz \
--pe1-1 D13_CKDL240013335-1A_22JMN3LT3_L3_1.P.qtrim.fq.gz --pe1-2 D13_CKDL240013335-1A_22JMN3LT3_L3_2.P.qtrim.fq.gz \
--pe1-fr -t $THREADS -m $MEM --cov-cutoff auto -o $DIR/rnaspade 

exit

```

### Other De novo Assembly evaluation
rnaQUAST 
GeneMarkS-T 

Additional reference 
```bash

# A) DNA sequences
# (Centruroides vittatus) https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_030686945.1/
TYPE=GCA
VERSION=1_ASM3068694v1
DATA=GCA_030686945.${VERSION}_genomic.fna.gz

URL=https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/030/686/945/GCA_030686945.1_ASM3068694v1/$DATA

#  -c, resume getting a partially-downloaded file.

wget -c $URL
```

