# Quantification step
## Using rsem

```bash
# 0) LOAD TOOLS
# git clone https://github.com/deweylab/RSEM.git
# make
# make install

EXPORT=/LUSTRE/apps/bioinformatica/RSEM/bin/
export PATH=$PATH:$EXPORT

EXPORT=/LUSTRE/apps/bioinformatica/bowtie2/bin/
export PATH=$PATH:$EXPORT

EXPORT=/LUSTRE/apps/bioinformatica/RSEM/bin/samtools-1.3/
export PATH=$PATH:$EXPORT
```
## Build index
```bash
sbatch bbuild.sh Trinity.fasta
```

## Bowtie alignment
```bash
for i in $(ls *_1.P.qtrim.fq.gz); do sbatch 02.bowtie2-alignment.sh Trinity.fasta $i; done

for i in $(ls *_1.P.qtrim.fq.gz); do sbatch 02.bowtie2-alignment.sh rnaspades.fasta $i; done

```

## Calculate expression (RSEM)
```bash 
sbatch rsem.sh transcripts.fasta
sbatch rsem.sh transcripts.fasta


```

```bash
# OUTPUT COUNT MATRIX TO DIFFERENTIAL EXPRESSION ANALYSIS\

ls -ltrh *isoforms.results
rsem-generate-data-matrix *isoforms.results > isoforms.counts.matrix


```
## Hisat alignment
```bash
sbatch hisat2build.sh transcripts.fasta
for i in $(ls *R1_p.fq.gz); do sbatch hisat2align.sh transcripts.fasta $i; done

for i in $(ls *_1.P.qtrim.fq.gz); do sbatch hisat2align.sh transcripts.fasta $i; done

```
## featureCount
```bash


export=/LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Software/subread-2.0.6-Linux-x86_64/bin
export PATH=$PATH:$EXPORT

EXPORT=/LUSTRE/bioinformatica_data/genomica_funcional/apps/gffread
export PATH=$PATH:$EXPORT

gffread rnaspades.fasta.transdecoder.gff3 -T -o rnaspades.fasta.transdecoder.gtf

srun featureCounts -a rnaspades.fasta.transdecoder.gtf -o feature_count.txt *bam 2> featureCount.log &

exit

featureCounts [options] -a <annotation_file> -o <output_file> input_file1 [input_file2] ...
featureCounts -a annotation.gtf -o counts.txt library1.bam library2.bam library3.bam


#Summarize multiple paired-end datasets:
featureCounts -p --countReadPairs -t exon -g gene_id -a annotation.gtf -o counts.txt library1.bam library2.bam library3.bam

#Summarize multiple datasets at the same time:
featureCounts -t exon -g gene_id -a annotation.gtf -o counts.txt library1.bam library2.bam library3.bam




```