 ```bash

EXPORT=/LUSTRE/apps/bioinformatica/TransDecoder-v5.7.0/
export PATH=$PATH:$EXPORT

EXPORT=/LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Software/eggnog-mapper-master
export PATH=$PATH:$EXPORT

EXPORT=/LUSTRE/bioinformatica_data/genomica_funcional/apps/tmhmm-2.0c/bin
export PATH=$PATH:$EXPORT

# Load depedencies for trinotate (such as blast, hmmpress, etc)
module load trinotate

 #* Initial creation of the Trinotate sqlite database and downloading of the required data sets
              
Trinotate --db sqlite.db --create --trinotate_data_dir TRINOTATE_DB



#  For future Trinotate execution, first set environment variable TRINOTATE_DATA_DIR like so:#
#  export TRINOTATE_DATA_DIR=/LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Annotation/TRINOTATE_DB

sbatch annotation.sh file.fasta

# Nucl. level
# ./diamond blastx -d DATABASE/uniprot_sprot -q cds.all.fa -o cds_all_vs_uniprot_sprot.blastx.tsv

# orf level:
#./diamond blastx -d DATABASE/uniprot_sprot -q pep.all.fa -o uniprot_sprot.ncbi.blastp.tsv

```