 ```bash

EXPORT=/LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Software
export PATH=$PATH:$EXPORT

EXPORT=/LUSTRE/apps/bioinformatica/TransDecoder-v5.7.0/
export PATH=$PATH:$EXPORT

EXPORT=/LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Software/eggnog-mapper-master
export PATH=$PATH:$EXPORT

EXPORT=/LUSTRE/bioinformatica_data/genomica_funcional/apps/tmhmm-2.0c/bin
export PATH=$PATH:$EXPORT

# Load depedencies for trinotate (such as blast, hmmpress, etc)
module load trinotate

 #* Initial creation of the Trinotate sqlite database and downloading of the required data sets
              
Trinotate --db sqlite.db --create --trinotate_data_dir TRINOTATE_DB



#  For future Trinotate execution, first set environment variable TRINOTATE_DATA_DIR like so:#
#  export TRINOTATE_DATA_DIR=/LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Annotation/TRINOTATE_DB

sbatch annotation.sh file.fasta

# Nucl. level
# ./diamond blastx -d DATABASE/uniprot_sprot -q cds.all.fa -o cds_all_vs_uniprot_sprot.blastx.tsv

# orf level:
#./diamond blastx -d DATABASE/uniprot_sprot -q pep.all.fa -o uniprot_sprot.ncbi.blastp.tsv

```

# Peptide search
Putative signal peptides and propeptides were predicted with the ProP 1.0 server [111] and SpiderP from Arachnoserver  (Cid-Uribe et al., 2019)

Arachnoserver
# https://arachnoserver.qfab.org/mainMenu.html
All spider toxin entries are sourced from UniProtKB/Swiss-Prot then manually curated by our expert team using available literature and patent information. Spider taxonomy is based on the latest version of the authoritative World Spider Catalog.
SpiderP predicts the location (or absence) of propeptides in spider toxins. The program first runs SignalP to search for signal peptides and then looks for propeptides in the remaining sequence using a support vector machine (SVM) algorithm. The program outputs any predicted signal and propeptide sequences as well as the predicted mature peptide region. If a propeptide is present, the residue number that is N-terminal to the propeptide cleavage site will also be reported. Please be aware that it might take some time for the SVM model to load.

 These data sets are joined with the assistance of the Sequence Retrieval System (SRS) ( 12 ) into a single non-redundant set containing peptide sequences, nucleotide sequences and protein structures (where available) for each toxin (https://doi.org/10.1093/nar/gkq1058)

```bash
diamond makedb --in all.pep.fa --db all.pep

diamond blastp -d /LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Centruroides/Reference/archnoDB/all.pep -q rnaspades.fasta.transdecoder.pep -p 20 -k 1 -e 1e-5 -o anchnoDB.diamond.blastp.outfmt6 --outfmt 6


diamond blastx -d /LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Centruroides/Reference/archnoDB/all.pep -q rnaspades.fasta -p 20 -k 1 -e 1e-5 -o anchnoDB.diamond.blastx.outfmt6 --outfmt 6
```
